src/applets/lock-keys/LockKeysApplet.vala

```diff
@@ -19,7 +19,10 @@ public class LockKeysApplet : Budgie.Applet {
 	Gtk.Box widget;
 	Gtk.Image caps;
 	Gtk.Image num;
+	Gtk.EventBox caps_box;
+	Gtk.EventBox num_box;
 	new Gdk.Keymap map;
+	string? xdotool;
```

Comment:
This doesn't actually need to be a class property, it can just be a variable limited to the scope of the `on_panel_widget_clicked` function, e.g. `string? xdotool = Environment.find_program_in_path("xdotool");`

Building on that, instead of having the nesting of the try / catch, we can just do:

``` vala
if ((xdotool == null) || (eventbutton.button != 1)) {
	return Gdk.EVENT_PROPAGATE;
}
```

Then have the try / catch

Reply:
Thanks.  For the "catch", on the off-chance of a spawn error, is it preferred to propagate or stop propagation?